src/book.js

```diff
@@ -310,20 +316,18 @@ export class Book {
         generateAggregate && !this.duplex
           ? await this.embedPagesInNewPdf(this.managedDoc, pdf1PageNumbers)
           : [null, null];
-      const forLoop = async () => {
-        for (let i = 0; i < this.rearrangedpages.length; i++) {
-          const signature = this.rearrangedpages[i];
-          await this.createsignatures({
-            embeddedPages: generateAggregate ? [embeddedPages0, embeddedPages1] : null,
-            aggregatePdfs: generateAggregate ? [aggregatePdf0, aggregatePdf1] : null,
-            pageIndexDetails: signature,
-            id: generateSignatures ? `signature${i}` : null,
-            isDuplex: this.duplex,
-            fileList: this.filelist,
-          });
-        }
-      };
-      await forLoop();
+
+      for (let i = 0; i < this.rearrangedpages.length; i++) {
```

Comment:
Trying to remember if there was a reason I didn't write this as `this.rearrangedpages.forEach()` - no obligation to make that change, though!

Reply:
changed